src/mapgen.cpp

```diff
@@ -1101,6 +1128,32 @@ class jmapgen_make_rubble : public jmapgen_piece {
     }
 };
 
+jmapgen_objects::jmapgen_objects( const int x_offset, const int y_offset, const int mapgensize )
+: x_offset( x_offset )
+, y_offset( y_offset )
+, mapgensize( mapgensize )
+{}
+
+bool jmapgen_objects::check_bounds( const jmapgen_place place, JsonObject &jso )
+{
+    if( place.x.val < 0 || (size_t)place.x.val > mapgensize - 1 ||
```

Comment:
This comparison is potentially dangerous: `(size_t)place.x.val > mapgensize - 1`. If `mapgensize` is zero (unlikely, but anyways), the condition is always `true` due to unsigned overflow.